server/src/main.rs

```diff
@@ -34,18 +38,42 @@ async fn main() -> Result<(), Error> {
     tracing_subscriber::fmt::init();
 
     let args = Args::parse();
-    let mut secret = None;
 
     capability::drop_unnecessary_capabilities()?;
 
-    if args.login {
+    let secret = if args.login {
         let password = rpassword::prompt_password("Enter the login password: ")?;
         if password.is_empty() {
             tracing::error!("Login password can't be empty.");
             return Err(Error::EmptyPassword);
         }
-        secret = Some(oo7::Secret::text(password));
-    }
+        Some(oo7::Secret::text(password))
+    } else if let Ok(credential_dir) = std::env::var("CREDENTIALS_DIRECTORY") {
+        // We try to unlock the login keyring with a system credential.
+        //
+        // FIXME We should not exit the service if the provided password is not
+        // the correct one in this branch. The credential might run out of sync
+        // and this is only a fallback.
```

Comment:
I guess this means we need a way to tell the server that this password might not be correct...

Reply:
Or just wrapping the secret in an `enum { Infallible(Secret), Fallible(secret) }` could be done to distinguish them and unlocking them with or without `?` depending on which enum was used.

Reply:
Doing something like that would be nice indeed.